### PR TITLE
Better numeric behaviors and support for shards

### PR DESCRIPTION
**Pull request by iSwepston**
- Adds support for shards on StatFacet 
- Adds ability to request percentiles in a stat facet
- Adds ability to generate numeric filters with a factory and apply them to a query
- Fixes some bugs in query validation of NumericStat in sharded environment
- Adds new test cases

### CODE REVIEW

File: zulia-client/src/test/java/io/zulia/client/WikiExamples.java [hunk 1]
```diff
@@ -689,4 +691,32 @@ public void queryingWithMapper(ZuliaWorkPool zuliaWorkPool, Mapper<Article> mapp
...
+		// Get the requested percentiles within 1% of their true value
+		search.addStat(new NumericStat("pubYear").setPercentiles(percentiles).setPercentilePrecision(0.01));
+		SearchResult searchResult = zuliaWorkPool.search(search);
+	}
+
+	public void numericFilters(ZuliaWorkPool zuliaWorkPool) throws Exception {
+		Search search = new Search("myIndexName");
+		search.addStat(new NumericStat("pubYear"));
+		// Search for pub years in range [2015, 2020]
+		search.addQuery(FilterFactory.rangeInt("pubYear").setMinValue(2015).setMinValue(2020).setEndpointBehavior(RangeBehavior.INCLUSIVE).toQuery());
```
Comment (mdavis95) [inline zulia-client/src/test/java/io/zulia/client/WikiExamples.java:713]: think this is supposed to say minValue / maxValue

File: zulia-server/src/main/java/io/zulia/server/node/ZuliaNode.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public void start() throws Exception {
 	}
 
 	public void start(boolean startREST) throws Exception {
-		LOG.info(getLogPrefix() + "starting");
+		LOG.info(getLogPrefix() + "starting release candidate");
```
Comment (mdavis95) [inline zulia-server/src/main/java/io/zulia/server/node/ZuliaNode.java:68]: remove debugging line

File: zulia-server/src/main/java/io/zulia/server/search/StatCombiner.java [hunk 3]
```diff
@@ -26,30 +37,314 @@ public int getShardIndex() {
...
+
+		// Send each group through the converter to generate a list
+		FacetStats.Builder globalStats = convertAndCombineFacetStats(globalStatsInternal);
+		List<FacetStats.Builder> facetStats = facetStatsGroups.values().stream().map(this::convertAndCombineFacetStats).toList();
+
+		// Find / bound errors in the sum element
+		for (FacetStats.Builder fs : facetStats) {
+			if (fs.getHasError()) {
+				// The error is the summation across all shards of the smallest value returned by each shard which did not return this facet
+				List<Integer> missing = getNonReportingShards(facetStatsGroups.get(fs.getFacet()));
```
Comment (mdavis95) [inline zulia-server/src/main/java/io/zulia/server/search/StatCombiner.java:96]: this is a great and readable way to do this

**Review by mdavis95 — APPROVED** [review]
This looks really nice.  Just a few random cleanups.

File: zulia-client/src/test/java/io/zulia/client/WikiExamples.java (hunk 1 above)
Comment (iSwepston) [reply zulia-client/src/test/java/io/zulia/client/WikiExamples.java:713]: Thanks yeah it should

File: zulia-server/src/main/java/io/zulia/server/search/StatCombiner.java (hunk 3 above)
Comment (iSwepston) [reply zulia-server/src/main/java/io/zulia/server/search/StatCombiner.java:96]: That section got re-written a bunch of times lol

**Comment by iSwepston** [comment]
Fixes pushed - ready to merge

**Pull request merged**